src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -136,46 +136,22 @@ class ArtifactRequirement:
     @classmethod
     def from_jvm_artifact_target(cls, target: Target) -> ArtifactRequirement:
         if not JvmArtifactFieldSet.is_applicable(target):
-            raise CoursierError(
+            raise AssertionError(
                 "`ArtifactRequirement.from_jvm_artifact_target()` only works on targets with "
                 "`JvmArtifactFieldSet` fields present."
             )
-
-        group = target[JvmArtifactGroupField].value
-        if not group:
-            raise InvalidTargetException(
-                f"The `group` field of {target.alias} target {target.address} must be set."
-            )
-
-        artifact = target[JvmArtifactArtifactField].value
-        if not artifact:
-            raise InvalidTargetException(
-                f"The `artifact` field of {target.alias} target {target.address} must be set."
-            )
-
-        version = target[JvmArtifactVersionField].value
-        if not version:
-            raise InvalidTargetException(
-                f"The `version` field of {target.alias} target {target.address} must be set."
-            )
-
-        url = target[JvmArtifactUrlField].value
-        jar_ = target[JvmArtifactJarSourceField]
-        jar = jar_ if jar_.value else None
-
-        if url and url.startswith("file:"):
-            raise CoursierError(
-                "Pants does not support `file:` URLS. Instead, use the `jar` field to specify the "
-                "relative path to the local jar file."
-            )
-
-        if url and jar:
-            raise CoursierError(
-                "You cannot specify both a `url` and `jar` for the same `jvm_artifact`."
-            )
```

Comment:
Now done with the Target API via `Target.validate()`. The major benefit of that is the validation is eager, `./pants list ::` will check it. We generally have preferred eager errors.

The downside is that this is less reslient to creating a new `JvmArtifact`-like target in a plugin, they might not have implemented `Target.validate` too.